src/CodeGen_WebGPU_Dev.cpp

```diff
@@ -0,0 +1,385 @@
+        std::string print_assignment(Type t, const std::string &rhs) override;
+
+        void visit(const Cast *) override;
+        void visit(const IntImm *) override;
+        void visit(const UIntImm *) override;
+        void visit(const For *) override;
+        void visit(const Min *op) override;
+        void visit(const Max *op) override;
+    };
+
+    std::ostringstream src_stream;
+    string cur_kernel_name;
+    CodeGen_WGSL wgsl;
+};
+
+CodeGen_WebGPU_Dev::CodeGen_WebGPU_Dev(const Target &t)
+    : wgsl(src_stream, t) {
+}
+
+void CodeGen_WebGPU_Dev::add_kernel(Stmt s,
+                                    const string &name,
+                                    const vector<DeviceArgument> &args) {
+    debug(2) << "CodeGen_WebGPU_Dev::add_kernel " << name << "\n";
+    debug(2) << "CodeGen_WebGPU_Dev:\n"
+             << s;
+
+    cur_kernel_name = name;
+    wgsl.add_kernel(s, name, args);
+}
+
+void CodeGen_WebGPU_Dev::init_module() {
+    debug(2) << "WebGPU device codegen init_module\n";
+
+    // Wipe the internal shader source.
+    src_stream.str("");
+    src_stream.clear();
+
+    // Write out the Halide math functions.
+    src_stream
+        << "fn float_from_bits(x : i32) -> f32 {return bitcast<f32>(x);}\n";
+}
+
+vector<char> CodeGen_WebGPU_Dev::compile_to_src() {
+    string str = src_stream.str();
+    debug(1) << "WGSL shader:\n"
+             << str << "\n";
+    vector<char> buffer(str.begin(), str.end());
+    buffer.push_back(0);
+    return buffer;
+}
+
+string CodeGen_WebGPU_Dev::get_current_kernel_name() {
+    return cur_kernel_name;
+}
+
+void CodeGen_WebGPU_Dev::dump() {
+    std::cerr << src_stream.str() << "\n";
+}
+
+string CodeGen_WebGPU_Dev::print_gpu_name(const string &name) {
+    return name;
+}
+
+string CodeGen_WebGPU_Dev::CodeGen_WGSL::print_name(const string &name) {
+    string new_name = c_print_name(name);
+
+    // The double-underscore prefix is reserved in WGSL.
+    if (new_name.length() > 1 && new_name[0] == '_' && new_name[1] == '_') {
+        new_name = "v" + new_name;
+    }
+
+    // TODO: Remove this when WGSL no longer requires buffers to be structures.
+    if (allocations.contains(name)) {
+        return new_name + ".data";
+    }
+
+    return new_name;
+}
+
+string CodeGen_WebGPU_Dev::CodeGen_WGSL::print_type(Type type,
+                                                    AppendSpaceIfNeeded space) {
+    ostringstream oss;
+
+    if (type.lanes() != 1) {
+        switch (type.lanes()) {
+        case 2:
+        case 3:
+        case 4:
+            oss << "vec" << type.lanes() << "<";
+            break;
+        default:
+            user_error << "Unsupported vector width in WGSL: " << type << "\n";
+        }
+    }
+
+    if (type.is_float()) {
+        user_assert(type.bits() == 32) << "WGSL only supports 32-bit floats";
+        oss << "f32";
+    } else if (type.bits() == 1) {
+        oss << "bool";
```

Comment:
This test should be `type.is_uint() && type.bits() == 1`, or, more simply, `type == UInt(1)`

Reply:
Done. I actually used `type == Bool()`, which is even simpler again.